iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -1137,11 +1140,7 @@ public void preDropPartitions(org.apache.hadoop.hive.metastore.api.Table hmsTabl
       ExprNodeDesc exprNodeDesc = SerializationUtilities
           .deserializeObjectWithTypeInformation(partExpr.getRight(), true);
       SearchArgument sarg = ConvertAstToSearchArg.create(conf, (ExprNodeGenericFuncDesc) exprNodeDesc);
-      for (PredicateLeaf leaf : sarg.getLeaves()) {
-        if (leaf.getColumnName() != null && !partitionFieldMap.containsKey(leaf.getColumnName())) {
-          throw new UnsupportedOperationException("Drop Partition not supported on Transformed Columns");
-        }
-      }
+      validatePartitionSpec(sarg, icebergTbl);
```

Comment:
@ayushtkn, can we drop `partitionFieldMap` it's no longer used

Reply:
done